DESCRIPTION

```diff
@@ -0,0 +1,21 @@
+Package: NetCoupler
+Title: Inference of Causal Links Between Metabolomics and Disease Incidence
+Version: 0.0.0.9000
+Authors@R: c(
+    person("Clemens", "Wittenbecher", email = NULL, role = c("aut", "cre")),
+    person("Luke", "Johnston", email = "lwjohnst@gmail.com", role = "aut")
+    )
+Description: Need to fill in here.
```

Comment:
The NetCoupler-algorithm was developed to identify direct effects of correlated, high-dimensional exposures on time-to-event data. 
In step 1 the implementation of the PC-algorithm in the pcalg package [1] is applied to estimate the skeleton of the underlying directed acyclic graph (DAG) that generated the complex exposure data (e.g., metabolomics data). 
In step 2, for each network variable (e.g., metabolite) the adjacency set (all direct neighbors) is extracted from this undirected network. Assuming complete coverage of the measurements, the Markov parents of the network variable are necessarily a subset of its direct network neighbors. Adjustment for the Markov parents is sufficient to block all confounding influences by other network-variables (d-separation). Adjusting for descendants should be avoided because it can potentially introduce collider bias. 
In step 3, a multi-model procedure is applied. The relation of each network variable with time-to-the event of interest is adjusted for every possible subset of direct neighbors. Thereby, a confidence set of estimates is generated, which necessarily contains the valid direct effect estimate. 
In step 4, network variables are classified based on the bounds of this confidence set of possible direct effects. As default, network variables are classified as directly affecting the outcome only if the confidence set of possible direct effects contains exclusively significant and consistent effect estimates. 
In step 5, the multi-model procedure is looped with adjusting all models for the previously identified direct effects (because these are potential confounders but cannot be colliders) until no further direct effects are identified.

A similar procedure is provided to identify direct influences of external factors on network variables.

1.   Markus Kalisch, Martin Maechler, Diego Colombo, Marloes H. Maathuis, Peter Buehlmann (2012). Causal Inference Using Graphical Models with the R Package pcalg. Journal of Statistical Software, 47(11), 1-26. URL http://www.jstatsoft.org/v47/i11/.